### PR TITLE
Classifier - TESS Light Curve Viewer - allow adding Annotations (ver 2)

### LINKED ISSUE #241: TESS LCV - viewer should be able to add annotations
## Required Feature

Package: lib-classifier
Project: Light Curve Viewer / Planet Hunters TESS
Requires: #240 to distinguish between Move (navigate) mode and Annotate mode.

The TESS Light Curve Viewer requires the ability to add Annotations.

**Desired Functionality**
- [ ] When the Classifier is in 'Move/Navigate' mode, the LCV will treat all mouse actions as navigation actions, as is usual with the D3 'zoom' behaviour.
- When the Classifier is in Annotate mode,
  - [ ] a click on the InterfaceLayer will create a new (confirmed) Annotation (a column with minX-maxX coordinates) centred around the click.
  - [ ] a drag on the InterfaceLayer will create a new ACTIVE Annotation, with minX starting at the point where the mouse click starts, and maxX at the point where mouse drag ends. (or vice versa if dragging right-to-left)
    - Releasing the drag will change the Active Annotation into a Confirmed Annotation
- Annotations must follow these rules:
  - [ ] Annotations that are currently being worked on are called Active Annotations
    - Active Annotations exist only in the D3 environment (and not the MobX store).
    - Active Annotations are initially removed from the Annotations Store, and re-added when they become Confirmed Annotations. (i.e. when the user finishes adding a new Annotation, or finishes editing an old Annotation.)
    - Rationale: If we don't do this, we'd need to actively change the value of a specific annotation in the array in the Annotations Store while the UI interactions happens and the Annotation is marked "active", which can get tricky, object reference-wise.
- [ ] Annotations might need to exist on their own AnnotationLayer that sits "above" (after) the InterfaceLayer. _(Note: this was originally proposed to make the annotation logic distinct from other forms of interaction, but I'm starting to realise that they may be too closely tied together... so regard this ToDo point with a pinch of salt.)_

### Dev Notes

Required: the ability to map mouse click coordinates to data point coordinates. See Roger's `plotly-prototype` code, notably:
```
import * as d3 from 'd3'
 export default function getXFromEvent (context, xScale) {
  const coords = d3.mouse(context)
  return Math.round(xScale.invert(coords[0]))
}
```

NOTE: for the TESS project, we assume the active Task is a Drawing Task of type Column, which is a very very specific assumption for how the LCV should be set up.

NOT PLANNED FOR: what happens when the Light Curve Viewer is used in conjunction with a Task that ISN'T a Drawing Task of type Column?

NOT PLANNED FOR: what if a user cancels out of Annotation mode and into Move/Navigation mode... while there exists an Active Annotation?

### PR DESCRIPTION
**Pull request by shaunanoordin**
## PR Overview
Package: `lib-classifier`
~~Requires (and is a sequel to) #279 and #352~~ 👌 (Merged!)
Replaces #309 (309 has gone waaay off base)
Closes #241 

The goal of this PR is to let users add annotations via the LCV
- [x] Users should be able to add new annotations in Annotate Mode.
  - [x] When in Annotate Mode, mouse click should add an annotation at the point of click.
  - [x] Mouse drag and wheel scroll should be ignored. 

screen shot 2018-12-04 at 15 45 09
_When in Annotation mode (selected from the Classifier page toolbar), users can click on the light curve to plonk down a "Graph Range"-type annotation (with a fixed width), which indicates the range where a dip in the brightness has been detected_

NOTE: the annotation data should be in the form of: `{ x: number, width: number }`
- `x` marks the centre of a selection range (aka a column) indicating a dip in the light curve.
- `width` marks the total range of the, er, selection range.

NOTE: not covered in this PR -
- ability to edit or remove existing Annotations.

### Dev Notes

See https://github.com/zooniverse/front-end-monorepo/pull/309#issuecomment-444182527 for full, rambling dev notes on...
1. Connecting the LightCurveViewer to the MobX stores
2. Using the "graph2dRangeX" workflow task

TL;DR: I need to make the LCV ignore any attempted annotation actions (e.g. adding/removing annotations) UNLESS the Current Workflow Task is "graph2dRangeX"

### Status
WIP - I still need to address a few comments from @rogerhutchings from #309 and check on the Current Workflow Task === 'graph2dRangeX'

### CODE REVIEW

**Comment by shaunanoordin** [comment]
Thanks @rogerhutchings! 👍 

- I'll get a new PR up for the Graph2dRangeXTask for the lib-classifier's TaskArea, that should be up next. (Also: this new PR should prevent any Annotation actions when a Graph2dRangeXTasks is NOT active in the task area.)
- Noted on the usage of D3 brushes - I mentioned in [309](https://github.com/zooniverse/front-end-monorepo/pull/309#issuecomment-453089540) that I need to revisit the UI interaction design (for adding annotations) when I work on the Edit/Remove PR, so that's next _next_ on my list.
